### DISCUSSION TITLE
Watermarking

### DISCUSSION THREAD

**Discussion by helmutkaufmann** [wintercms/winter#1042] [General]
I would like to watermark images with a string, where the color of the string should be the inverse of color of the underlying pixel. So, if it is the pixel of an image is RGB 0000FF, the watermarking pixel would be FFFF00. How is that done (efficiently)?

**Comment by LukeTowers** [comment]
@helmutkaufmann that would be pretty jarring, why are you wanting that specifically? Why not overlay a watermark with some sort of translucent background so that it's readable on anything?

Theoretically someone could bypass the watermark by just pre-populating the image with the exact inverse of the current pixel values in the exact location where your watermark would end up, thus when your code goes to apply your "watermark" all it actually does is fix the uploaded image rendering an image with no watermark.

**Reply by helmutkaufmann** [reply]
I am doing a lot of b/w images. and for Murphy's law, I always end up with a white text on the white part of the image for some of the images...

**Comment by helmutkaufmann** [comment]
Let's close this. Thanks @LukeTowers .
